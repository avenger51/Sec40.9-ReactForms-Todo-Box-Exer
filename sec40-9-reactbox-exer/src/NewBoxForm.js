import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';

function NewBoxForm({ addBox }) {
  const [formData, setFormData] = useState({
    width: '',
    height: '',
    backgroundColor: ''
  });

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  };

  const handleSubmit = e => {
    e.preventDefault();
    addBox({ ...formData, id: uuidv4() });
    setFormData({ width: '', height: '', backgroundColor: '' });
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor='width'>Width:</label>
      <input 
        id='width'
        type='text' 
        name='width' 
        value={formData.width} 
        onChange={handleChange} 
      />
      <label htmlFor='height'>Height:</label>
      <input 
        id='height'
        type='text' 
        name='height' 
        value={formData.height} 
        onChange={handleChange} 
      />
      <label htmlFor='backgroundColor'>Background Color:</label>
      <input 
        id='backgroundColor'
        type='text' 
        name='backgroundColor' 
        value={formData.backgroundColor} 
        onChange={handleChange} 
      />
      <button>Add a new box</button>
    </form>
  );
}

export default NewBoxForm;
