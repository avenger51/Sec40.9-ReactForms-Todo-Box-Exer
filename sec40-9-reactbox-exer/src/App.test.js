import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Box from './Box';
import NewBoxForm from './NewBoxForm';

//test('renders BoxList component', () => {
//  render(<App />);
//  const linkElement = screen.getByText('boxListTestId');
//  expect(linkElement).toBeInTheDocument();
//});


test('renders App component without crashing', () => {
  render(<App />);
});


test('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Box component without crashing', () => {
  render(<Box id="1" width="100" height="100" backgroundColor="red" removeBox={() => {}} />);
});



test('Box component matches snapshot', () => {
  const tree = renderer.create(<Box id="1" width="100" height="100" backgroundColor="red" removeBox={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});


test('form submission adds a new box', () => {
  const addBoxMock = jest.fn();
  render(<NewBoxForm addBox={addBoxMock} />);

  fireEvent.change(screen.getByLabelText(/width/i), { target: { value: '100' } });
  fireEvent.change(screen.getByLabelText(/height/i), { target: { value: '100' } });
  fireEvent.change(screen.getByLabelText(/background color/i), { target: { value: 'red' } });
  fireEvent.click(screen.getByText(/add a new box/i));

  expect(addBoxMock).toHaveBeenCalledWith(expect.objectContaining({
    width: '100',
    height: '100',
    backgroundColor: 'red'
  }));
});
