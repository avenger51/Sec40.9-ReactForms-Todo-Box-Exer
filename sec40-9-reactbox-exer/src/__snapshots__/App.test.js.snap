// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component matches snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "red",
        "height": "100px",
        "width": "100px",
      }
    }
  />
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;

exports[`matches snapshot 1`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="backgroundColor"
      >
        Background Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button>
        Add a new box
      </button>
    </form>
  </div>
</div>
`;
